src/main/java/com/google/sps/data/AggregationResponseEntry.java

```diff
@@ -40,7 +42,12 @@ public AggregationResponseEntry(
       }
     }
 
-    this.count = deviceIds.size();
-    this.deviceIds = deviceIds;
+    this.count = devices.size();
+    this.deviceIds = devices.stream()
+        .map(device -> device.getDeviceId())
+        .collect(Collectors.toList());
+    this.serialNumbers = devices.stream()
+        .map(device -> device.getSerialNumber())
+        .collect(Collectors.toList());
```

Comment:
while this might be a bit cleaner, iterating over the list twice seems excessive